timesketch/lib/aggregators/vega.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Vega Spec aggregation.
+
+This file provides a manual vega spec aggregator. This is useful for
+analyzers that generate specific vega specs (or altair charts) that
+are not yet supported by other aggregators and/or charts. This can
+also be used by those manually generating charts in notebooks.
+
+This analyzer is not useful for UI and therefore is "hidden" from
+views.
+"""
+from __future__ import unicode_literals
+
+import altair as alt
+import pandas as pd
+
+from timesketch.lib.aggregators import manager
+from timesketch.lib.aggregators import interface
+
+
+class VegaResult:
+    """Result object for manual vega specification aggregations.
+
+    Attributes:
+        chart_type: Chart type to render, defaults to "table".
+    """
+
+    def __init__(self, spec):
+        """Initialize the object.
+
+        Args:
+            spec: dict with the vega specification.
+        """
+
+        self.chart_type = 'manual_vega'
+        self._spec = spec
+
+    # pylint: disable=unused-argument
+    def to_dict(self, encoding=False):
+        """Encode aggregation result as dict.
+
+        Args:
+            encoding: Boolean indicating if encoding info should be returned.
+
+        Returns:
+            Dict with aggregation result.
+        """
+        datasets = self._spec.get('datasets', {})
+        values = []
+
+        for dataset in datasets.values():
+            values.extend(list(dataset))
+
+        return {
```

Comment:
maybe use dict() here

Reply:
so you prefer this to be 

```
return dict(values=values)
```

rather than

```
return {
    'values': values,
}
```
???

Reply:
Kind of, but no strong opinion :) Feel free to keep.